ee/tuf/generator/generator.go

```diff
@@ -0,0 +1,153 @@
+		flOutputDirs   = flagset.String("output-dirs", "./assets/tuf,../../pkg/packaging/assets/tuf", "comma-separated output directories")
+	)
+	if err := flagset.Parse(os.Args[1:]); err != nil {
+		level.Error(logger).Log("msg", "error parsing flags", "err", err)
+		os.Exit(1) //nolint:forbidigo // Fine to use os.Exit outside of launcher proper
+	}
+
+	// relevel with the debug flag
+	logger = logutil.NewCLILogger(*flDebug)
+
+	// Validate required flags
+	missingOpt := false
+	for f, val := range map[string]string{
+		"tuf-url":       *flTufURL,
+		"metadata-path": *flMetadataPath,
+		"output-dirs":   *flOutputDirs,
+	} {
+		if val == "" {
+			level.Error(logger).Log("msg", "Missing required flag", "flag", f)
+			missingOpt = true
+		}
+	}
+	if missingOpt {
+		os.Exit(1) //nolint:forbidigo // Fine to use os.Exit outside of launcher proper
+	}
+
+	ctx := context.Background()
+
+	if err := updateTUFMetadata(ctx, logger, *flTufURL, *flMetadataPath, *flOutputDirs); err != nil {
+		level.Error(logger).Log("msg", "error updating TUF metadata", "err", err)
+		os.Exit(1) //nolint:forbidigo // Fine to use os.Exit outside of launcher proper
+	}
+}
+
+func updateTUFMetadata(ctx context.Context, logger log.Logger, tufURL, metadataPath, outputDirsStr string) error {
+	level.Info(logger).Log("msg", "Starting TUF metadata update", "url", tufURL)
+
+	// Create a temporary directory to store TUF metadata
+	tempDir, err := os.MkdirTemp("", "tuf-update")
+	if err != nil {
+		return fmt.Errorf("creating temp directory: %w", err)
+	}
+	defer os.RemoveAll(tempDir)
+
+	level.Debug(logger).Log("msg", "Created temporary directory", "path", tempDir)
+
+	// Set up the local store
+	localStore, err := filejsonstore.NewFileJSONStore(tempDir)
+	if err != nil {
+		return fmt.Errorf("initializing local TUF store: %w", err)
+	}
+
+	// Set up the remote store
+	remoteOpts := client.HTTPRemoteOptions{
+		MetadataPath: metadataPath,
+	}
+
+	level.Debug(logger).Log(
+		"msg", "Configuring remote TUF store",
+		"url", tufURL,
+		"metadata_path", metadataPath,
+	)
+
+	remoteStore, err := client.HTTPRemoteStore(tufURL, &remoteOpts, http.DefaultClient)
+	if err != nil {
+		return fmt.Errorf("initializing remote TUF store: %w", err)
+	}
+
+	// Initialize the TUF client with our initial root
+	tufClient := client.NewClient(localStore, remoteStore)
+	level.Debug(logger).Log("msg", "Initializing TUF client with initial root")
+
+	if err := tufClient.Init(initialRootJSON); err != nil {
+		return fmt.Errorf("initializing TUF client: %w", err)
+	}
+
+	// Update the root metadata to the latest version
+	level.Info(logger).Log("msg", "Updating TUF metadata to latest version")
+
+	_, err = tufClient.Update()
+	if err != nil {
+		return fmt.Errorf("updating TUF metadata: %w", err)
+	}
+
+	// Get the latest root.json
+	rootPath := filepath.Join(tempDir, "root.json")
+	latestRoot, err := os.ReadFile(rootPath)
+	if err != nil {
+		return fmt.Errorf("reading updated root.json: %w", err)
+	}
```

Comment:
I would probably pull this right from `localStore` instead -- you should be able to do this --

```suggestion
	metadata, err := localStore.GetMeta()
	if err != nil {
		return fmt.Errorf("reading updated metadata from local store: %w", err)
	}
	latestRoot = metadata["root.json"]
```

Reply:
Gah, sorry about the `latestRoot = metadata["root.json"]` / `latestRoot := metadata["root.json"]` typo!